runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1606,7 +1608,41 @@ MM_CopyForwardScheme::copyForwardCollectionSet(MM_EnvironmentVLHGC *env)
 	/* Do any final work to regions in order to release them back to the main collector implementation */
 	postProcessRegions(env);
 
-	return copyForwardCompletedSuccessfully(env);
+	static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_abortFlagRaisedDuringPGC = copyForwardCompletedSuccessfully(env);
+}
+
+#if defined(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD)
+void
+MM_CopyForwardScheme::concurrentCopyForwardCollectionSet(MM_EnvironmentVLHGC *env)
+{
+	/* isConcurrentCycleInProgress() tells us if this is the first PGC increment or not */
+	if (!isConcurrentCycleInProgress())
+	{
+		copyForwardPreProcess(env);
+	}
+
+	/* And perform the copy forward. Temporary as this will be updated to call concurrent copy forward state machine */
```

Comment:
you may add that along this step isConcurrentCycleInProgress may/will flip/change its value.